transports/dns/src/lib.rs

```diff
@@ -21,21 +21,16 @@
 //! # [DNS name resolution](https://github.com/libp2p/specs/blob/master/addressing/README.md#ip-and-name-resolution)
 //! [`Transport`] for libp2p.
 //!
-//! This crate provides the type [`async_std::Transport`] and [`tokio::Transport`]
-//! for use with `async-std` and `tokio`,
-//! respectively.
+//! This crate provides the type [`tokio::Transport`]
+//! for use with `tokio`,
 //!
 //! A [`Transport`] is an address-rewriting [`libp2p_core::Transport`] wrapper around
 //! an inner `Transport`. The composed transport behaves like the inner
 //! transport, except that [`libp2p_core::Transport::dial`] resolves `/dns/...`, `/dns4/...`,
 //! `/dns6/...` and `/dnsaddr/...` components of the given `Multiaddr` through
 //! a DNS, replacing them with the resolved protocols (typically TCP/IP).
 //!
-//! The `async-std` feature and hence the [`async_std::Transport`] are
-//! enabled by default. Tokio users can furthermore opt-in
-//! to the `tokio-dns-over-rustls` and `tokio-dns-over-https-rustls`
-//! features. For more information about these features, please
-//! refer to the documentation of [trust-dns-resolver].
+//! The `tokio` feature and hence the [`tokio::Transport`] are enabled by default.
 //!
```

Comment:
```suggestion
//! Alternative runtimes or resolvers can be used though a manual implementation of [`Resolver`].
```
This depends on the outcome of https://github.com/libp2p/rust-libp2p/pull/5955#discussion_r2048540910. But we agree on making the abstractions public then we should also remove the `docs(hidden)` the `Resolver` trait so that downstream crates can use it more easily. 
```